lib/mix/ecto.ex

```diff
@@ -97,6 +97,19 @@ defmodule Mix.Ecto do
   end
 
   @doc """
+  Ensures the given repository's migrations path exists on the filesystem.
+  """
+  @spec ensure_migrations_path(Ecto.Repo.t) :: Ecto.Repo.t | no_return
+  def ensure_migrations_path(repo) do
+    path = Path.relative_to(migrations_path(repo), Mix.Project.app_path)
```

Comment:
Just use migrations_path(repo) and use Path.relative_to_cwd when building the error message. :)